allauth/account/forms.py

```diff
@@ -44,7 +44,7 @@ def _make_hash_value(self, user, timestamp):
             EmailAddress.objects
             .filter(user=user)
             .values_list('email', flat=True))
-        ret += '|'.join(sorted(emails))
+        ret += '|'.join(sorted((str(e) for e in emails)))
```

Comment:
I would rather see the issue addressed on this stop (instead of using the str() workaround):

 `emails = set([user.email] if user.email else [])`

Reply:
@pennersr  Done!